### PR TITLE
added endpoint that filters data using java streams

### PR DESCRIPTION
**Pull request by vglalitha**
Java Streams

### CODE REVIEW

File: dropwizard-demo/src/main/java/com/dep/Services/TwitterImplement.java [hunk 1]
```diff
@@ -49,25 +54,50 @@ public Status sendTweets(String tweet) throws TwitterException {
...
+                String message = status.getText();
+                createdAt =status.getCreatedAt();
+                Format format = new SimpleDateFormat("dd-mm-yyy HH:mm:ss");
+                String date = format.format(createdAt);
+                String twitterHandle = status.getUser().getScreenName();
+                tweetRespons = new TweetRespons(message,name,twitterHandle,profileImageUrl,date);
+                arrayList.add(tweetRespons);
+            }
+        } catch (TwitterException e) {
+        }
```
Comment (anasuya1117) [inline dropwizard-demo/src/main/java/com/dep/Services/TwitterImplement.java:95]: Apply DRY , Empty catch block ?  add a log

File: dropwizard-demo/src/main/java/com/dep/models/TweetRespons.java [hunk 2]
```diff
@@ -10,7 +10,7 @@ public TweetRespons(String message, String twitterHandle,String name, String pro
         this.createdAt = createdAt;
     }
 
-    public TweetRespons() {
+    public TweetRespons(String message, String twitterHandle, String name) {
```
Comment (anasuya1117) [inline dropwizard-demo/src/main/java/com/dep/models/TweetRespons.java:13]: Spelling mistake `TweetRespons`

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java [hunk 3]
```diff
@@ -29,11 +31,23 @@ public Controller() {
...
         return Response.ok(twitterImplement.fetchLatestTweets()).build();
     }
 
+
+    @GET
+    @Path("/filteredTweets")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response filteredTweets(@QueryParam("searchKey") String searchKey) throws TwitterException{
+        List<TweetRespons>  response ;
+        response = twitterImplement.getFilteredTweets(searchKey) ;
```
Comment (anasuya1117) [inline dropwizard-demo/src/main/java/com/dep/resource/Controller.java:45]: merge these lines.

File: dropwizard-demo/test/com/test/TwitterImplementTest.java [hunk 4]
```diff
@@ -66,23 +87,22 @@ public void testCase_sendTweet_successCase() throws TwitterException {
 
     @Test
     public void testCase_fetchTweet_successCase() throws TwitterException {
+        ArrayList<TweetRespons> expectedlist = mock(ArrayList.class);
```
Comment (anasuya1117) [inline dropwizard-demo/test/com/test/TwitterImplementTest.java:90]: Good

**Review by anasuya1117 — COMMENTED** [review]
Add more mock data for test and try to cover more scenarios.

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java [hunk 5]
```diff
@@ -29,11 +31,20 @@ public Controller() {
     }
 
     @GET
-    @Path("GetTweets")
-    public static Response fetchTweets() {
+    @Path("/GetTweets")
```
Comment (anasuya1117) [inline dropwizard-demo/src/main/java/com/dep/resource/Controller.java:34]: URI's should be lowercase

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java [hunk 6]
```diff
@@ -29,11 +31,20 @@ public Controller() {
...
+    public Response fetchTweets() {
         return Response.ok(twitterImplement.fetchLatestTweets()).build();
     }
 
+    @GET
+    @Path("/filteredTweets")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response filteredTweets(@QueryParam("searchKey") String searchKey) throws TwitterException {
+        List<TweetResponse> response;
+        response = twitterImplement.getFilteredTweets(searchKey);
```
Comment (anasuya1117) [inline dropwizard-demo/src/main/java/com/dep/resource/Controller.java:44]: merge these two lines

File: dropwizard-demo/src/main/java/com/dep/Services/TwitterImplement.java [hunk 7]
```diff
@@ -42,32 +42,41 @@ public Status sendTweets(String tweet) throws TwitterException {
...
             }
         } catch (TwitterException e) {
         }
         return arrayList;
     }
+
+    public List<TweetResponse> getFilteredTweets(String tweets) {
+        ArrayList<TweetResponse> listTweets = new ArrayList<>();
+        List<TweetResponse> filteredTweets;
+        List<Status> statuses = null;
```
[the remark this answers is not in the document]
Comment (vglalitha) [reply dropwizard-demo/src/main/java/com/dep/Services/TwitterImplement.java]: addressed the review

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java (hunk 5 above)
Comment (vglalitha) [reply dropwizard-demo/src/main/java/com/dep/resource/Controller.java]: addressed the review

File: dropwizard-demo/src/main/java/com/dep/resource/Controller.java (hunk 6 above)
Comment (vglalitha) [reply dropwizard-demo/src/main/java/com/dep/resource/Controller.java]: addressed the review

File: dropwizard-demo/src/main/java/com/dep/Services/TwitterImplement.java [hunk 8]
```diff
@@ -42,32 +44,37 @@ public Status sendTweets(String tweet) throws TwitterException {
     }
 
 
-    public ArrayList<TweetRespons> fetchLatestTweets() {
-
-        ArrayList<TweetRespons> arrayList = new ArrayList<>();
-        List<Status> statuses = null;
+    public ArrayList<TweetResponse> fetchLatestTweets() {
+        ArrayList<TweetResponse> TweetList = new ArrayList<>();
```
Comment (PoojaMohite-Khoros) [inline dropwizard-demo/src/main/java/com/dep/Services/TwitterImplement.java:48]: Variable name doesn't start with capital letter

Comment (PoojaMohite-Khoros) [reply dropwizard-demo/src/main/java/com/dep/Services/TwitterImplement.java:48]: https://confluence.dev.lithium.com/display/DEV/09+-+Naming+Conventions please read and correct all variable names, class names, file names etc

File: dropwizard-demo/src/main/java/com/dep/Services/TwitterImplement.java [hunk 9]
```diff
@@ -42,32 +44,37 @@ public Status sendTweets(String tweet) throws TwitterException {
...
         } catch (TwitterException e) {
+            logger.error("error in fetching tweets");
         }
-        return arrayList;
+        return TweetList;
+    }
+
+    public List<TweetResponse> getFilteredTweets(String tweets) {
+        ArrayList<TweetResponse> listTweets = fetchLatestTweets();
+        int len=tweets.length();
```
Comment (PoojaMohite-Khoros) [inline dropwizard-demo/src/main/java/com/dep/Services/TwitterImplement.java:72]: here align code properly 

cmd + options +shift + L , when pressed at the same time, u will get code alignment window , run it please

**Review by PoojaMohite-Khoros — CHANGES_REQUESTED** [review]

File: dropwizard-demo/src/main/java/com/dep/Services/TwitterImplement.java (hunk 8 above)
Comment (vglalitha) [reply dropwizard-demo/src/main/java/com/dep/Services/TwitterImplement.java]: addressed the review

File: dropwizard-demo/src/main/java/com/dep/Services/TwitterImplement.java (hunk 9 above)
Comment (vglalitha) [reply dropwizard-demo/src/main/java/com/dep/Services/TwitterImplement.java]: addressed the review

**Review by PoojaMohite-Khoros — APPROVED** [review]
We can always improve the filter.
Please merge and create PR for next lab

**Pull request merged**